src/python/pants/engine/internals/parser.py

```diff
@@ -419,15 +420,17 @@ def parse(
             code = compile(build_file_content, filepath, "exec", dont_inherit=True)
             exec(code, global_symbols)
         except NameError as e:
-            valid_symbols = sorted(s for s in global_symbols.keys() if s != "__builtins__")
-            original = e.args[0].capitalize()
+            frame = traceback.extract_tb(e.__traceback__, limit=-1)[0]
+            msg = e.args[0][0].upper() + e.args[0][1:]
```

Comment:
it wasn't clear to me that this turns "name 'dne' is not defined" -> "Name 'dne' is not defined"
```suggestion
            msg = e.args[0][0].upper() + e.args[0][1:]  # capitalise first letter of NameError message
```